yarl/_url.py

```diff
@@ -291,6 +292,17 @@ def __str__(self):
         val = self._val
         if not val.path and self.is_absolute() and (val.query or val.fragment):
             val = val._replace(path="/")
+        if (port := self._get_port()) is None:
+            # https://datatracker.ietf.org/doc/html/rfc3986.html#section-6.2.3
```

Comment:
Additionally to a link, it'd be nice to have a summary text right above it. As in, “how this branch is to be interpreted”.